helix-core/src/comment.rs

```diff
@@ -53,6 +53,26 @@ pub fn toggle_line_comments(doc: &Rope, selection: &Selection, token: Option<&st
 
         changes.reserve((end - start).saturating_sub(skipped.len()));
 
+        // determine margin of 0 or 1 for uncommenting; if any comment token is not followed by a space,
+        // a margin of 0 is used for all lines.
+        let mut margin = 1;
+        if commented {
+            for line in lines.clone() {
```

Comment:
Is it possible to not iterate over this twice?

Reply:
Merging with the bottom loop would cause a margin of `1` to be used until a line that should use a margin of `0`.

Reply:
However, as @cessen suggested, I'm going to change the margin calculation to occur in `find_line_comment`.